ui/store/actions.js

```diff
@@ -3273,6 +3273,19 @@ export function setRecoveryPhraseReminderLastShown(lastShown) {
   };
 }
 
+export function setOutdatedBrowserWarningLastShown(lastShown) {
+  return async (dispatch) => {
+    try {
+      await submitRequestToBackground('setOutdatedBrowserWarningLastShown', [
+        lastShown,
+      ]);
+      await forceUpdateMetamaskState(dispatch);
+    } catch (error) {
+      dispatch(displayWarning(error.message));
+    }
```

Comment:
This `displayWarning` method should be considered deprecated as well. That behavior shows the error directly to the error in a very confusing and nonsensical way.

Sorry when I suggested using `try..catch` I hadn't considered whether there was an error to catch in the first place. It's probably better to let this one bubble up.

```suggestion
    await submitRequestToBackground('setOutdatedBrowserWarningLastShown', [
      lastShown,
    ]);
    await forceUpdateMetamaskState(dispatch);
```

Reply:
Gotcha, I appreciate you explaining the rationale behind it too, thanks!

Reply:
Documented here: https://github.com/MetaMask/metamask-extension/pull/17465